cmd/interchaincmd/tokentransferrercmd/deploy.go

```diff
@@ -304,41 +343,11 @@ func CallDeploy(_ []string, flags DeployFlags) error {
 		}
 	}
 
-	var (
-		homeKey        string
-		homeKeyAddress string
-	)
-	if flags.homeFlags.homeAddress == "" {
-		genesisAddress, genesisPrivateKey, err := contract.GetEVMSubnetPrefundedKey(
-			app,
-			network,
-			flags.homeFlags.chainFlags,
-		)
-		if err != nil {
-			return err
-		}
-		homeKey, err = flags.homeFlags.privateKeyFlags.GetPrivateKey(app, genesisPrivateKey)
-		if err != nil {
-			return err
-		}
-		if homeKey == "" {
-			homeKey, err = prompts.PromptPrivateKey(
-				app.Prompt,
-				"pay for home deploy fees",
-				app.GetKeyDir(),
-				app.GetKey,
-				genesisAddress,
-				genesisPrivateKey,
-			)
-			if err != nil {
-				return err
-			}
-		}
-		pk, err := crypto.HexToECDSA(homeKey)
-		if err != nil {
-			return err
-		}
-		homeKeyAddress = crypto.PubkeyToAddress(pk.PublicKey).Hex()
+	// Get the key to be used to deploy the token home contract and collateralize the remote
+	// in the case that it is a native token remote.
+	homeKey, homeKeyAddress, err := getHomeKeyAndAddress(app, network, flags.homeFlags)
```

Comment:
previously the bug was not asking for it when needed, but now it may also ask for it when not needed,
as in the case the home preexists and there is no collateralization needed.